py/desispec/io/util.py

```diff
@@ -785,9 +785,15 @@ def get_tempfilename(filename):
     race condition protection (the last one do do os.rename wins, but at least
     different processes won't corrupt each other's files).
     """
-    base, extension = os.path.splitext(filename)
     pid = os.getpid()
-    tempfile = f'{base}_tmp{pid}{extension}'
+    if filename.endswith('.gz'):
+        filename, second_ext = os.path.splitext(filename)
+    elif filename.endswith('.fz'):
+        filename, second_ext = os.path.splitext(filename)
```

Comment:
Minor:  `filename.endswith( ('.gz', '.fz') )` would also work and avoids duplicating the logic of what to do when the file ends with .fz or .gz.  Note: `.endswith` works with str or tuple of str, but not list of str.

Reply:
This should be resolved in 638e35f